src/script/script_gui.cpp

```diff
@@ -898,8 +898,16 @@ struct ScriptDebugWindow : public Window {
 		ScriptLogTypes::LogData &log = this->GetLogData();
 		if (log.empty()) return;
 
-		Rect br = r.Shrink(WidgetDimensions::scaled.bevel);
-		Rect tr = r.Shrink(WidgetDimensions::scaled.framerect);
+		Rect fr = r.Shrink(WidgetDimensions::scaled.framerect);
+
+		DrawPixelInfo tmp_dpi;
+		if (!FillDrawPixelInfo(&tmp_dpi, fr)) return;
+		AutoRestoreBackup dpi_backup(_cur_dpi, &tmp_dpi);
+
+		/* Draw content (now coordinates given to DrawString* are local to the new clipping region). */
+		fr = fr.Translate(-fr.left, -fr.top);
+		fr.left -= this->hscroll->GetPosition();
```

Comment:
You can keep the clipping region with
```
tmp_dpi.left += fr.left;
tmp_dpi.top += fr.top;
```
after FillDrawPixelInfo, which might avoid coordinate confusion. (This is does NWidgetLeaf::Draw() does it.)

Reply:
I used code from `TextfileWindow::DrawWidget()` but I see `IndustryDirectoryWindow::DrawWidget()` uses your suggestion, and it indeed looks cleaner.